### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by ntietje1**
# Checklist

- [X] I have performed a self-review of my code
- [X] I have reached out to another developer to review my code
- [X] New and existing unit tests pass locally with my changes
- [X] All Lint and CI checks are passing

### CODE REVIEW

File: backend/internal/storage/postgres/semesters.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		"SELECT id, name, classroom_id, active, org_id FROM semesters WHERE org_id = $1",
+		orgID,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var semesters []models.Semester
+	for rows.Next() {
```
Comment (alexangione419) [inline backend/internal/storage/postgres/semesters.go:20]: Why are you manually doing this and not using pgx.CollectRows

**Review by alexangione419 — COMMENTED** [review]
Database/routes stuff looks solid, haven't looked at business logic yet

File: frontend/src/pages/Semesters/Creation/index.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
+import React, { useEffect, useState } from "react";
+import "./styles.css";
+import { useNavigate } from "react-router-dom";
+
```
Comment (CamPlume1) [inline frontend/src/pages/Semesters/Creation/index.tsx:4]: Break down into subcomponents, utilize useEffect for child component refreshes. This is hard to understand and maintain, and better to refactor now than later IMO

**Review by CamPlume1 — CHANGES_REQUESTED** [review]
See creation page

**Comment by ntietje1** [comment]
@CamPlume1 There seems to be an issue with the lint rules on the frontend. Not sure what is causing it to not like my types and therefore give me an error basically everywhere I use them. I'm going to put an ignore on the files for now since it works.

File: frontend/src/pages/Semesters/Creation/index.tsx (hunk 2 above)
Comment (ntietje1) [reply frontend/src/pages/Semesters/Creation/index.tsx]: fixed!

File: backend/internal/storage/postgres/semesters.go (hunk 1 above)
Comment (ntietje1) [reply backend/internal/storage/postgres/semesters.go]: fixed!

**Review by nandini-ghosh — APPROVED** [review]
yassss slayyy

**Review by alexangione419 — APPROVED** [review]
Bakcend looks good to me

**Review by CamPlume1 — APPROVED** [review]
looks good

**Pull request merged**